featuretools/entityset/entityset.py

```diff
@@ -593,6 +598,9 @@ def normalize_entity(self, base_entity_id, new_entity_id, index,
         elif make_time_index:
             # Create a new time index based on the base entity time index.
             base_time_index = base_entity.time_index
+            t_index = base_entity[base_time_index]
+            if not isinstance(t_index, (vtypes.NumericTimeIndex, vtypes.DatetimeTimeIndex)):
+                raise TypeError("{0} is not a NumericTimeIndex or DatetimeTimeIndex, but type {1}".format(base_time_index, type(t_index)))
```

Comment:
Let's start the error message with "Time index {0}" so there's more context about the error.  And at the end of the error message we can add a suggestion for fixing it by running `set_time_index` on the entity.

I think we should make this a sanity check we run any time `normalize_entity` is called since if this gets triggered the entity is definitely misconfigured